### DISCUSSION TITLE
Is it possible to have basePath with a dynamic segment (locale)?

### DISCUSSION THREAD

**Discussion by szimek** [vercel/next.js#26852] [Help]
Hi!

We've got a blog that currently uses Gatsby, but because the build times are pretty long (over 6K pages, ~10 minutes), we're considering switching to Next.js.

Our blog is served from `/:locale/blog/` subpath and we've got another app running at `/:locale/`. Currently, Gatsby is configured to put all its own generated asset files (js, json, css etc.) into `./public/assets/blog` folder. We've got a rule in our nginx server to proxy all requests to both `/:locale/blog/*` and `/assets/blog/*` to our blog app.

Is it possible to do something similar with Next.js? In general, I want to make sure that all files generated by Next.js are under a unique path that I can easily proxy all requests to.

I think this question can also be asked this way: when using `basePath` and i18n,  is the URL structure `/:locale/:basePath/*` or `/:basePath/:locale/*` and what's the path to asset files (webpack runtime etc.) in this case?

**Comment by robertwbradford** [comment]
I would love to see something like the following proposed syntax when `basePath` is used with i18n routing: https://github.com/vercel/next.js/discussions/17078#discussioncomment-220722
